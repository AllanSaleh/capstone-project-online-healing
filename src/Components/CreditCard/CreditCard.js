import React from 'react';
import C1 from './C1.svg';
import C2 from './C2.svg';
import C3 from './C3.svg';
import MasterLogo from './MasterLogo.svg';
import VisaLogo from './VisaLogo.svg';

let selectedBackground;
let selectedLogo;
let chosenClass;

export default function CreditCard({
  background,
  logo,
  date,
  number,
  name,
  id,
  setID,
  select,
  deletable,
}) {

  if (background === 1) selectedBackground = C1;
  if (background === 2) selectedBackground = C2;
  if (background === 3) selectedBackground = C3;

  if (logo === 'Master') selectedLogo = MasterLogo;
  if (logo === 'Visa') selectedLogo = VisaLogo;

  if (select)
    chosenClass =
      'border-8 border-blue-dark rounded-lg border-dashed absolute w-full cursor-pointer';
  else
    chosenClass =
      'border-8 border-transparent rounded-lg border-dashed absolute w-full cursor-pointer';

  return (
    <div
      onClick={(e) => setID(e.target.closest('.relative').id)}
      id={id}
      className="relative inline-block w-2/6 h-80 box-border mx-12"
    >
      <img className={chosenClass} src={selectedBackground} alt="Background" />
      <img
        className="cursor-pointer absolute top-11 right-6"
        src={selectedLogo}
        alt="Logo"
      />
      <div className="cursor-pointer absolute top-12 left-12 text-white text-paragraph">
        {date}
      </div>
      <div className="cursor-pointer absolute top-32 left-12 text-white text-paragraph">
        {number}
      </div>
      <div className="cursor-pointer absolute top-52 left-12 text-white text-paragraph">
        {name}
      </div>
      {deletable && (
        <button
          type="button"
          className="absolute bottom-9 right-5 h-12 w-28 text-subtitle text-white bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
        >
          DELETE
        </button>
      )}
    </div>
  );
}
