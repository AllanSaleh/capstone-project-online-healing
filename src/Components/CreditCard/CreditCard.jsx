import React from 'react';
import firebase from 'firebase';

import C1 from './Images/C1.svg';
import C2 from './Images/C2.svg';
import C3 from './Images/C3.svg';
import MasterLogo from './Images/MasterLogo.svg';
import VisaLogo from './Images/VisaLogo.svg';

export default function CreditCard({
  background,
  logo,
  date,
  number,
  name,
  id,
  setID,
  select,
  deletable,
  loginStatus,
}) {
  const DeleteCard = (cardID) => {
    let selectedCard;
    firebase
      .firestore()
      .collection('users')
      .doc('cUld5Z0ytjTuTrbeu95n')
      .get()
      .then((doc) => {
        doc.data().cards.forEach((card) => {
          if (card.number === cardID) selectedCard = card;
        });
        firebase
          .firestore()
          .collection('users')
          .doc(loginStatus.user_id)
          .update({
            cards: firebase.firestore.FieldValue.arrayRemove(selectedCard),
          });
        window.location.reload();
      });
  };

  return (
    <div
      onClick={(e) => setID(e.target.closest('.relative').id)}
      id={id}
      className="relative inline-block w-full lg:w-1/3 h-40 lg:h-80 box-border mx-2 lg:mx-12"
    >
      <img
        className={`border-8 rounded-lg border-dashed absolute top-4 lg:top-0 w-full cursor-pointer ${
          select ? 'border-blue-dark' : 'border-transparent'
        }`}
        src={background === 1 ? C1 : background === 2 ? C2 : C3}
        alt="Background"
      />
      <img
        className="w-8 lg:w-auto cursor-pointer absolute top-8 right-4 lg:top-11 lg:right-6"
        src={logo === 'Master' ? MasterLogo : VisaLogo}
        alt="Logo"
      />
      <div className="cursor-pointer absolute top-8 left-4 lg:top-12 lg:left-12 text-white text-sm lg:text-paragraph">
        {date}
      </div>
      <div className="cursor-pointer absolute top-16 left-4 lg:top-32 lg:left-12 text-white text-sm lg:text-paragraph">
        {number}
      </div>
      <div className="cursor-pointer absolute top-24 left-4 lg:top-52 lg:left-12 text-white text-sm lg:text-paragraph">
        {name}
      </div>
      {deletable && (
        <button
          onClick={(e) => DeleteCard(e.target.closest('.relative').id)}
          type="button"
          className="absolute bottom-7 right-4 h-6 w-12 lg:bottom-9 lg:right-4 lg:h-12 lg:w-28 text-xs lg:text-subtitle text-white bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
        >
          DELETE
        </button>
      )}
    </div>
  );
}
