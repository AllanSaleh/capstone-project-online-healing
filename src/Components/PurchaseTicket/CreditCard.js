import React from 'react';
import C1 from './C1.svg';
import C2 from './C2.svg';
import C3 from './C3.svg';
import MasterLogo from './MasterLogo.svg';
import VisaLogo from './VisaLogo.svg';

let backgrounds;
let logos;

export default function CreditCard({ background, logo, date, number, name }) {
  if (background === 1) backgrounds = C1;
  if (background === 2) backgrounds = C2;
  if (background === 3) backgrounds = C3;

  if (logo === 'Master') logos = MasterLogo;
  if (logo === 'Visa') logos = VisaLogo;

  return (
    <div className="relative inline-block w-2/6 h-80 box-border mx-12">
      <img
        className="absolute w-full h-full"
        src={backgrounds}
        alt="Background"
      />
      <img className="absolute top-10 right-6" src={logos} alt="Logo" />
      <div className="absolute top-14 left-4 text-white text-paragraph">
        {date}
      </div>
      <div className="absolute top-28 left-4 text-white text-paragraph">
        {number}
      </div>
      <div className="absolute top-40 left-4 text-white text-paragraph">
        {name}
      </div>
      <button
        type="button"
        className="absolute bottom-8 right-4 h-12 w-32 text-subtitle bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
      >
        Delete -
      </button>
    </div>
  );
}
