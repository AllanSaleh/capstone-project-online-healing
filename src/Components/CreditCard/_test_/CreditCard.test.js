import * as React from 'react';
import CreditCard from "../CreditCard";
import * as renderer from "react-test-renderer";

it('Credit Card renders correctly', () => {
  const tree = renderer
    .create(<CreditCard/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});