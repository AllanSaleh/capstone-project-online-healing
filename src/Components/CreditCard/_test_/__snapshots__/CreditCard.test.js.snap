// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Credit Card renders correctly 1`] = `
<div
  className="relative inline-block w-full lg:w-1/3 h-40 lg:h-80 box-border mx-2 lg:mx-12"
  onClick={[Function]}
>
  <img
    alt="Background"
    className="border-8 rounded-lg border-dashed absolute top-4 lg:top-0 w-full cursor-pointer border-transparent"
    src="C3.svg"
  />
  <img
    alt="Logo"
    className="w-8 lg:w-auto cursor-pointer absolute top-8 right-4 lg:top-11 lg:right-6"
    src="VisaLogo.svg"
  />
  <div
    className="cursor-pointer absolute top-8 left-4 lg:top-12 lg:left-12 text-white text-sm lg:text-paragraph"
  />
  <div
    className="cursor-pointer absolute top-16 left-4 lg:top-32 lg:left-12 text-white text-sm lg:text-paragraph"
  />
  <div
    className="cursor-pointer absolute top-24 left-4 lg:top-52 lg:left-12 text-white text-sm lg:text-paragraph"
  />
</div>
`;
